tests/test_simple_usage.py

```diff
@@ -0,0 +1,135 @@
+import numpy as np
+import pytest
+import tensorflow as tf
+
+from sklearn.preprocessing import OneHotEncoder
+
+from scikeras.utils import loss_name
+from scikeras.wrappers import KerasClassifier, KerasRegressor
+
+
+N_CLASSES = 4
+FEATURES = 8
+n_eg = 100
+X = np.random.uniform(size=(n_eg, FEATURES)).astype("float32")
+
+
+def shallow_net(single_output=False, loss=None, in_dim=FEATURES, compile=False):
+    model = tf.keras.Sequential()
+    model.add(tf.keras.layers.Input(shape=(in_dim,)))
+    model.add(tf.keras.layers.Dense(in_dim, activation="sigmoid"))
+
+    if single_output:
+        model.add(tf.keras.layers.Dense(1))
+    else:
+        model.add(tf.keras.layers.Dense(N_CLASSES))
+
+    if compile:
+        model.compile(loss=loss)
+
+    return model
+
+
+@pytest.mark.parametrize(
+    "loss",
+    [
+        "binary_crossentropy",
+        "categorical_crossentropy",
+        "sparse_categorical_crossentropy",
+        "poisson",
+        "kl_divergence",
+        "hinge",
+        "categorical_hinge",
+        "squared_hinge",
+    ],
+)
+def test_user_compiled(loss):
+    """Test to make sure that user compiled classification models work with all
+    classification losses.
+    """
+    model__single_output = True if "binary" in loss else False
+    if loss == "binary_crosentropy":
+        y = np.random.randint(0, 2, size=(n_eg,))
+    elif loss == "categorical_crossentropy":
+        # SciKeras does not auto one-hot encode unless
+        # loss="categorical_crossentropy" is explictily passed to the constructor
+        y = np.random.randint(0, N_CLASSES, size=(n_eg, 1))
+        y = OneHotEncoder(sparse=False).fit_transform(y)
+    else:
+        y = np.random.randint(0, N_CLASSES, size=(n_eg,))
+    est = KerasClassifier(
+        shallow_net,
+        model__compile=True,
```

Comment:
``` python
@pytest.mark.parametrize("compile", [True, False])
```

?

Reply:
It might be worth collapsing these tests. They're all very similar, and I'm having a hard time telling the difference.

Reply:
I attempted to collapse these tests. Let me know if it is clearer now.